echopop/inversion/inversion_matrix.py

```diff
@@ -0,0 +1,1230 @@
+        - nasc: NASC values at reference frequency
+        - number_density: Estimated areal number density (individuals/m²)
+        - biomass_density: Estimated areal biomass density (kg/m²)
+        - Biological parameters: length_mean, radius_mean, body_volume,
+          body_density, body_weight, etc.
+
+    Notes
+    -----
+    The function assumes organisms can be modeled as uniformly bent cylinders
+    for volume estimation. Body weight calculation uses:
+
+    .. math::
+        W = \\rho_{sw} \\cdot g \\cdot \\pi r^2 L
+
+    where W is body weight, ρ_sw is seawater density, g is the relative
+    density factor, r is mean radius, and L is mean length.
+    """
+
+    # Unpackage the parameters using optimized extraction
+    parameters = _extract_parameters_optimized(inverted_data)
+
+    # Compute the average body radius for a single animal
+    parameters["radius_mean"] = parameters["length_mean"] / parameters["length_radius_ratio"]
+
+    # Estimate the average volume assuming an uniformly bent cylinder [single animal]
+    parameters["body_volume"] = np.pi * parameters["radius_mean"] ** 2 * parameters["length_mean"]
+
+    # Compute the animal body density [single animal]
+    parameters["body_density"] = density_sw * parameters["g"]
+
+    # Compute average body weight [single animal]
+    parameters["body_weight"] = parameters["body_volume"] * parameters["body_density"]
+
+    # Subset the dataset further to the single frequency
+    acoustic_data = (
+        inverted_data.loc[
+            :, inverted_data.columns.get_level_values("frequency") == reference_frequency
+        ]
+    ).droplevel("frequency", axis=1)
+
+    # Repeat the same for the coordinates
+    reference_nasc = (
+        nasc_data.loc[:, nasc_data.columns.get_level_values("frequency") == reference_frequency]
+    )[reference_frequency].to_frame("nasc")
+
+    # Calculate the areal number density
+    if aggregate_method == "transect":
+        reference_nasc.loc[:, "number_density"] = invert_transect_number_density(
+            acoustic_data, reference_nasc, parameters
+        ).to_numpy()
+    else:
+        reference_nasc["number_density"] = invert_intervals_number_density(
+            acoustic_data, parameters
+        )
+
+    # Compute areal biomass density
+    reference_nasc["biomass_density"] = (
+        reference_nasc["number_density"] * parameters["body_weight"]
+    ).fillna(0.0)
+
+    # Find shared indices
+    idx = list(set(reference_nasc.index.names).intersection(parameters.index.names))
+
+    # Reset indices
+    reference_nasc = reference_nasc.reset_index().set_index(idx)
+
+    # Concatenate the parameters
+    return pd.concat(
+        [
+            reference_nasc,
+            parameters.reset_index()
+            .drop("number_density", axis=1)
+            .set_index(idx)
+            .reindex(reference_nasc.index),
+        ],
+        axis=1,
+    )
+
+
+def invert_transect_number_density(
```

Comment:
```suggestion
def calculate_transect_number_density(
```
since it is not an inversion.